tests/store/filters/test_status_application_version_filters.py

```diff
@@ -0,0 +1,322 @@
+from typing import List, Tuple, Iterable
+import datetime as dt
+
+from sqlalchemy.orm import Query
+
+from cg.store import Store
+from cg.store.filters.status_application_version_filters import (
+    filter_application_versions_by_application_id,
+    filter_application_versions_by_application,
+    filter_application_versions_before_valid_from,
+    filter_application_versions_by_version,
+    order_application_versions_by_valid_from_desc,
+    apply_application_versions_filter,
+    ApplicationVersionFilter,
+)
+from cg.store.models import Application, ApplicationVersion
+
+from tests.store.api.conftest import (
+    fixture_applications_store,
+)
+from tests.store_helpers import StoreHelpers
+
+
+def test_filter_application_version_by_application_correct_application(
+    store_with_different_application_versions: Store,
+    helpers: StoreHelpers,
+):
+    """Test that the application of the filtered application version is the correct one."""
+    # GIVEN a store with different applications
+    applications: List[Application] = store_with_different_application_versions.get_applications()
+    linked_application: Application = applications[0]
+    unlinked_application: Application = applications[-1]
+    assert unlinked_application != linked_application
+
+    # GIVEN that one application has a linked application version in the store
+    app_version_query: Query = store_with_different_application_versions._get_query(
+        table=ApplicationVersion
+    )
+    assert app_version_query.first().application == linked_application
+
+    # WHEN filtering an application version query by the linked application
+    filtered_app_version_query: Query = filter_application_versions_by_application(
+        application_versions=app_version_query,
+        application=linked_application,
+    )
+
```

Comment:
This assertion is covered by the last assertion: If `filtered_app_version_query` is not a valid query it will not be able to call the `.all()` function and an error would be raised